File: internal/domain/resume.go
package domain

// StructuredResume 解析后存 Redis key: resume:{user_id}
type StructuredResume struct {
	UserID string `json:"user_id"`
	// Name / Phone / Email：手动填表由前端写入；
	// PDF 解析路径暂不填充（后续可在 LLM prompt 中补充提取）。
	Name        string             `json:"name,omitempty"`
	Phone       string             `json:"phone,omitempty"`
	Email       string             `json:"email,omitempty"`
	Skills      []string           `json:"skills"`
	Projects    []ResumeProject    `json:"projects"`
	Internships []ResumeInternship `json:"internships"`
	Education   ResumeEducation    `json:"education"`
}

type ResumeProject struct {
	Name        string   `json:"name"`
	TechStack   []string `json:"tech_stack"`
	Description string   `json:"description"`
	Highlights  []string `json:"highlights"`
}

type ResumeInternship struct {
	Company     string `json:"company,omitempty"`
	Role        string `json:"role,omitempty"`
	Description string `json:"description,omitempty"`
}

type ResumeEducation struct {
	School     string `json:"school"`
	Major      string `json:"major"`
	Graduation string `json:"graduation"`
}

File: frontend/src/types/resume.ts
// 简历相关类型定义

// 教育背景
export interface ResumeEducation {
  school: string;
  major: string;
  graduation: string;
}

// 项目经验
export interface ResumeProject {
  name: string;
  tech_stack: string[];
  description: string;
  highlights: string[];
}

// 实习经历
export interface ResumeInternship {
  company?: string;
  role?: string;
  description?: string;
}

// 结构化简历
export interface StructuredResume {
  user_id: string;
  // 手动填表场景使用；PDF 解析路径目前不填充
  name?: string;
  phone?: string;
  email?: string;
  skills: string[];
  projects: ResumeProject[];
  internships: ResumeInternship[];
  education: ResumeEducation;
}

// 简历提交请求（表单方式）
export interface ResumeSubmitRequest {
  user_id: string;
  name?: string;
  phone?: string;
  email?: string;
  skills: string[];
  projects: ResumeProject[];
  internships: ResumeInternship[];
  education: ResumeEducation;
}

// PDF 上传请求
export interface ResumePDFUploadRequest {
  user_id: string;
  file: File;
}

export type ResumeParseResponse = StructuredResume;
